modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -783,8 +788,41 @@ private void prepareAndExecuteOperation(CoreEvent event, Supplier<ExecutorCallba
 
     ExecutionContextAdapter<T> operationContext = oep.getExecutionContextAdapter();
 
-    executeOperation(operationContext, mapped(callbackSupplier.get(), operationContext,
-                                              isTargetWithPolicies(event) ? valueReturnDelegate : returnDelegate));
+    setCurrentEvent((PrivilegedEvent) event);
+    boolean wasProcessorPathSet = setCurrentLocation();
+    try {
+      executeOperation(operationContext, mapped(callbackSupplier.get(), operationContext,
+                                                isTargetWithPolicies(event) ? valueReturnDelegate : returnDelegate));
+    } finally {
+      unsetCurrentLocation(wasProcessorPathSet);
+    }
+  }
+
+  private boolean setCurrentLocation() {
+    if (MDC.get(PROCESSOR_PATH_MDC_KEY) != null) {
+      return false;
+    }
+
+    if (processorPath == null) {
```

Comment:
you have a potential race condition here with the first couple of events as processorPath needs to be set for the first time. I think that the processorPath field should be initialised on the `initialise()` phase.